independent-projects/tools/devtools-testing/src/test/java/io/quarkus/devtools/project/create/NoOfferringsTest.java

```diff
@@ -0,0 +1,133 @@
+                .build();
+
+        enableRegistryClient();
+
+    }
+
+    protected String getMainPlatformKey() {
+        return MAIN_PLATFORM_KEY;
+    }
+
+    @Test
+    public void allEnabled() throws Exception {
+        var catalog = ExtensionCatalogResolver.builder().build().resolveExtensionCatalog();
+        assertThat(catalog.getExtensions()).hasSize(1);
+        final Map<ArtifactKey, Set<ArtifactCoords>> extensionMap = toExtensionMap(catalog);
+        assertThat(extensionMap).containsEntry(
+                ArtifactKey.fromString("org.acme:acme-quarkus-other"),
+                Set.of(ArtifactCoords.pom("io.quarkus", "quarkus-bom", "1.1.2")));
+    }
+
+    @Test
+    public void stream20() throws Exception {
+        try {
+            ExtensionCatalogResolver.builder().build().resolveExtensionCatalog(PlatformStreamCoords.fromString("2.0"));
+            fail("Should have failed");
+        } catch (RegistryResolutionException e) {
+            assertThat(e).hasMessage("Couldn't find any extension catalog for stream 2.0");
+        }
+    }
+
+    @Test
+    public void quarkusVersion222() throws Exception {
+        try {
+            ExtensionCatalogResolver.builder().build().resolveExtensionCatalog("2.2.2");
+            fail("Should have failed");
+        } catch (RegistryResolutionException e) {
+            assertThat(e).hasMessage("Quarkus extension registry registry.acme.org did not provide any extension catalog");
+        }
+    }
+
+    @Test
+    public void stream10() throws Exception {
+        try {
+            ExtensionCatalogResolver.builder().build().resolveExtensionCatalog(PlatformStreamCoords.fromString("1.0"));
+            fail("Should have failed");
+        } catch (RegistryResolutionException e) {
+            assertThat(e).hasMessage("Couldn't find any extension catalog for stream 1.0");
+        }
+    }
+
+    @Test
+    public void quarkusVersion112() throws Exception {
+        var catalog = ExtensionCatalogResolver.builder().build().resolveExtensionCatalog("1.1.2");
+        assertThat(catalog.getExtensions()).hasSize(1);
+        final Map<ArtifactKey, Set<ArtifactCoords>> extensionMap = toExtensionMap(catalog);
+        assertThat(extensionMap).containsEntry(
+                ArtifactKey.fromString("org.acme:acme-quarkus-other"),
+                Set.of(ArtifactCoords.pom("io.quarkus", "quarkus-bom", "1.1.2")));
+    }
+
+    private static Map<ArtifactKey, Set<ArtifactCoords>> toExtensionMap(ExtensionCatalog catalog) {
+        final Map<ArtifactKey, Set<ArtifactCoords>> extensionMap = new HashMap<>(catalog.getExtensions().size());
+        for (Extension e : catalog.getExtensions()) {
+            final Set<ArtifactCoords> boms = new HashSet<>(e.getOrigins().size());
+            for (ExtensionOrigin o : e.getOrigins()) {
+                boms.add(o.getBom());
+            }
+            extensionMap.put(e.getArtifact().getKey(), boms);
+        }
+        return extensionMap;
+    }
```

Comment:
This would be useful as an utility method inside `ExtensionCatalog`

Reply:
Also, the test class name has an extra `r`, just noticed 😉

Reply:
There is also the possibility of writing it using Streams, but the actual is okay: 

```java
return catalog.getExtensions().stream()
                .collect(Collectors.toMap(
                        e -> e.getArtifact().getKey(),
                        e -> e.getOrigins().stream().map(ExtensionOrigin::getBom).collect(Collectors.toSet()))
                );
```